### PR TITLE
Cplusplus fixups

### PR DESCRIPTION
**Pull request by erik-johnson-nimbelink**
A handful of fixes to libraries to make life in C++ land a little easier.

### CODE REVIEW

**Comment by erik-johnson-nimbelink** [comment]
I don't have access to the Jenkins build server's report, but it looks like it failed on the C++ guards. Is Jenkins currently able to build the master branch? It's baffling to me that it'd fail on something so basic, especially since so much other nRF code has those same guards already...

**Comment by rlubos** [comment]
Hi, we're sorting out the CI now, should get a full report soon enough.

Anyway, I can already see it will complain about the commit messages, we follow the Zephyr convention here:
https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines

In general, add `--signoff` entry, some commit body (the commits are simple, so can be the commit body) and a proper title, no longer than 72 characters. And I'd appreciate if you could use the following prefixes in the commit titles:
```
lib: bsdlib:
lib: lte_lc:
net: lib: download_client: 
```

Otherwise, the code seems fine, so I don't expect any issues there.

**Comment by erik-johnson-nimbelink** [comment]
Okay, sounds good! Would you want a force-push update with the commit message changes? (Assuming GitHub handles force pushes with Pull Requests well.)

**Comment by rlubos** [comment]
Yes, you'll need to force-push after amending the commits.

**Comment by thst-nordic** [comment]
- You can ignore the `checkpatch `warnings.
 - `gitlint` checks for these 4 items:   (you are missing #2) 
          https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines

**Comment by erik-johnson-nimbelink** [comment]
Sorry, thought rlubos meant that the sign-off line would count as the "body" of the commit. I've updated the commits to include a little bit of the why of the commits.

File: include/net/download_client.h [hunk 1]
```diff
@@ -60,20 +60,23 @@ enum download_client_evt_id {
 	DOWNLOAD_CLIENT_EVT_DONE,
 };
 
+struct download_fragment {
+    const void *buf;
```
Comment (rlubos) [inline include/net/download_client.h]: nit: Please indent with tabulation. Here and one line below.

Comment (erik-johnson-nimbelink) [reply include/net/download_client.h]: Ack, sorry, thought I had tabs for everything.

**Pull request merged**